Source/Route.swift

```diff
@@ -16,35 +16,31 @@
 
 import UIKit
 
-public class Route {
+class Route {
     
-    static let routeParameterPattern = ":[a-zA-Z0-9-_]+"
-    static let urlParameterPattern = "([^/]+)"
+    enum Pattern: String {
+        case RouteParam = ":[a-zA-Z0-9-_]+"
+        case UrlParam = "([^/]+)"
+    }
     
-    let routeParameter: NSRegularExpression = try! NSRegularExpression(pattern: routeParameterPattern, options: .CaseInsensitive)
-    let urlParameter: NSRegularExpression = try! NSRegularExpression(pattern: urlParameterPattern, options: .CaseInsensitive)
+    let routeParameter: NSRegularExpression = try! NSRegularExpression(pattern: Pattern.RouteParam.rawValue, options: .CaseInsensitive)
+    let urlParameter: NSRegularExpression = try! NSRegularExpression(pattern: Pattern.UrlParam.rawValue, options: .CaseInsensitive)
     
     // parameterized route, ie: /video/:id
-    public var route: String! {
-        didSet {
-            if let regex = regex() {
-                routePattern = regex
-            }
-        }
-    }
+    let route: String
     
     // route in its regular expression pattern, ie: /video/([^/]+)
-    var routePattern: String!
+    var routePattern: String?
     
     // url params found in route
-    var urlParamKeys: [String] = [String]()
+    var urlParamKeys = [String]()
     
     init(aRoute: String) {
-        setupRoute(aRoute)
-    }
-    
-    private func setupRoute(aRoute: String) {
         route = aRoute
+        guard let regex = regex() else {
+            fatalError("\(route) was unable to converted to a regex")
```

Comment:
we should probably use swift 2.0 error handling here and handle the error with a state

```
enum RoutingError: ErrorType {
  case RegexConversion(route: String)
  ...
  var info: String {
    switch self {
      case RegexParseError(let string):
        return "/(string) was unable to converted to a regex"
    }
  }
}
```

```
init(route: String) throws {
  self.route = route
  guard let regex = regex() else {
    throw RoutingError.RegexConversion
  }
  ...

```